accepted/2021/json-polymorphism.md

```diff
@@ -0,0 +1,275 @@
+# System.Text.Json polymorphic serialization
+
+**Owner** [Eirik Tsarpalis](https://github.com/eiriktsarpalis)
+
+This documents describes the proposed design for extending [polymorphism support](https://github.com/dotnet/runtime/issues/45189) in System.Text.Json.
+
+[Draft Implementation PR](https://github.com/dotnet/runtime/pull/53882).
+
+## Background
+
+By default, System.Text.Json will serialize a value using a converter derived from its declared type,
+regardless of what the runtime type of the value might be. This behavior is in line with the
+Liskov substitution principle, in that the serialization contract is unique (or "monomorphic") for a given type `T`,
+regardless of what subtype of `T` we end up serializing at runtime.
+
+A notable exception to this rule is members of type `object`, in which case the runtime type of the value
+is looked up and serialization is dispatched to the converter corresponding to that runtime type.
+This is an instance of _polymorphic serialization_, in the sense that the schema might vary depending on
+the runtime type a given `object` instance might have. 
+
+Conversely, in _polymorphic deserialization_ the runtime type of a deserialized value might vary depending on the
+shape of the input encoding. Currently, System.Text.Json does not offer any form of support for polymorphic 
+deserialization.
```

Comment:
"System.Text.Json does not offer any form of support for polymorphic 
deserialization."

It may be useful to explain that a System.Object property is materialized with either a `JsonElement` or a `JsonNode`. Although not polymorphic in the true sense, it does fill the feature gap (and provide a workaround) to be able to deserialize arbitrary JSON that may be aligned to a polymorphic model.

Reply:
Makes sense, added clarification.